lib/spack/spack/util/module_cmd.py

```diff
@@ -164,10 +164,11 @@ def get_path_arg_from_module_line(line):
     else:
         path_arg = line.split()[2]
 
-    if not os.path.exists(path_arg):
-        tty.warn("Extracted path from module does not exist:"
-                 "\n\tExtracted path: " + path_arg +
-                 "\n\tFull line: " + line)
+    for dir in path_arg.split(':'):
```

Comment:
@balay instead of this, could you move the original check down to `get_path_from_module`? At that point, a single path should have been extracted. i.e. replace

```
return get_path_from_module_contents(text, mod)
```

with

```
p = get_path_from_module_contents(text, mod)
if not os.path.exists(p):
  tty.warn(...)
return p
```

Reply:
```
balay@cori10:~/spack> git diff |cat
diff --git a/lib/spack/spack/util/module_cmd.py b/lib/spack/spack/util/module_cmd.py
index 69a48306a..2e88398ad 100644
--- a/lib/spack/spack/util/module_cmd.py
+++ b/lib/spack/spack/util/module_cmd.py
@@ -144,11 +144,6 @@ def get_path_arg_from_module_line(line):
         path_arg = words_and_symbols[-2]
     else:
         path_arg = line.split()[2]
-
-    if not os.path.exists(path_arg):
-        tty.warn("Extracted path from module does not exist:"
-                 "\n\tExtracted path: " + path_arg +
-                 "\n\tFull line: " + line)
     return path_arg
 
 
@@ -162,8 +157,11 @@ def get_path_from_module(mod):
     # Read the module
     text = modulecmd('show', mod, output=str, error=str).split('\n')
 
-    return get_path_from_module_contents(text, mod)
-
+    p = get_path_from_module_contents(text, mod)
+    if not os.path.exists(p):
+        tty.warn("Extracted path from module does not exist:"
+                 "\n\tExtracted path: " + p)
+    return p
 
 def get_path_from_module_contents(text, module_name):
     tty.debug("Module name: " + module_name)
balay@cori10:~/spack> spack env m4 >/dev/null
balay@cori10:~/spack> spack env mfem >/dev/null
balay@cori10:~/spack> 
```
Is this what you mean? It works for me.. [as in not printing wrong warning. But I don't know how to test an actual error PATH - that should give a warning]

Reply:
> Is this what you mean?

yes

> I don't know how to test an actual error PATH

Does this interfere with a test? The warning should be printed in the same circumstances. The way I implemented it originally didn't make it easy to test anyhow, and it should only come up if there is an internal Spack issue with processing it, so I agree it would be hard to test manually, so I wouldn't require that either.

Reply:
The patch works for my usage. So will update the PR now